src/hotspot/share/opto/phaseX.cpp

```diff
@@ -61,6 +61,36 @@ NodeHash::NodeHash(Arena *arena, uint est_max_size) :
   memset(_table,0,sizeof(Node*)*_max);
 }
 
+//-----------------------------------------------------------------------------
+
+bool NodeHash::have_equivalent_inputs(const Node* n, const Node* k) const {
+  // For commutative operations with same controlling edge
+  // perform order agnostic input edge comparison to promote
+  // node sharing.
+  uint req = n->req();
+  // Predicated vector operations are sensitive to ordering of inputs.
+  // When the mask corresponding to a vector lane is false then
+  // the result of the operation is corresponding lane of its first operand.
+  //   i.e. RES = VEC1.lanewise(OPER, VEC2, MASK) is semantically equivalent to
+  //        RES = VEC1.BLEND(VEC1.lanewise(OPER, VEC2), MASK)
+  if (n->is_commutative_vector_operation() && !n->is_predicated_vector()) {
```

Comment:
Wait, so any of the commutative operations can also be predicated? Can you tell me what that does?
`VEC1.BLEND(VEC1.lanewise(OPER, VEC2), MASK)`
What does this do? Can you make an example?

And why would it not be ok to check if the `VEC1` and `VEC2` are the same (with swapping), and that the `MASK` is identical?